src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -7123,8 +7126,11 @@ static void Main()
 
             CreateCompilation(source).VerifyDiagnostics(
                 // (23,9): error CS0121: The call is ambiguous between the following methods or properties: 'C.Test(Func<(Func<short>, int)>)' and 'C.Test(Func<(Func<byte>, int)>)'
+                //         Test(()=>(()=>(byte)1, 1));
+                Diagnostic(ErrorCode.ERR_AmbigCall, "Test").WithArguments("C.Test(System.Func<(System.Func<short>, int)>)", "C.Test(System.Func<(System.Func<byte>, int)>)").WithLocation(23, 9),
```

Comment:
How is this not a breaking change?

Reply:
Technically, it is a breaking change caused by a bug that I fixed in type inference involving lambdas. It was deeply broken before, in a way that could cause bad code generation in the presence of a switch expression. I don't have an easy way to keeping it working without breaking things. It is pretty hard to hit the issue.

---
In reply to: [301778341](https://github.com/dotnet/roslyn/pull/37052#discussion_r301778341) [](ancestors = 301778341)